aya/src/programs/mod.rs

```diff
@@ -533,3 +579,24 @@ impl_try_from_program!(
     CgroupSkb,
     LircMode2
 );
+
+pub struct ProgramInfo(bpf_prog_info);
+
+impl ProgramInfo {
+    pub fn name(&self) -> Option<String> {
```

Comment:
Shouldn't this return Option<&[u8]>? It's valid for the name not to be utf8, I think callers should (optionally) do the conversion here

Reply:
> Shouldn't this return &[u8]? It's valid for the name not to be utf8, I think callers should (optionally) do the conversion here

I just realised that we use String for program names everywhere else so I'm leaving this up to you, I'm ok with either

Reply:
So we have both of them now as an option: fn name() which gives you `&[u8]` and fn name_as_str() which returns `Option<&str>`.